### PR TITLE
Change Travis CI config to use newly presented build stages feature

### LINKED ISSUE #28: [TODO] Change Travis CI config to use newly presented build stages feature
Refs:
* https://blog.travis-ci.com/2017-05-11-introducing-build-stages
* https://docs.travis-ci.com/user/build-stages#Examples

*Note: I'm going to move PYPI deployment to separate stage, so it would wait for tests in all envs to complete.*

### PR DESCRIPTION
**Pull request by webknjaz**
Closes #28.

### CODE REVIEW

**Comment by webknjaz** [comment]
[postponed]

Ref bug: https://github.com/travis-ci/beta-features/issues/11#issuecomment-302766298

**Comment by webknjaz** [comment]
Ref bug: travis-ci/travis-ci#7789

**Comment by webknjaz** [comment]
That bug has been fixed.

**Comment by webknjaz** [comment]
@jaraco what about moving `TOXENV=pre-commit` job to separate pre-flight check stage? It runs faster than others and will fail the build faster, preventing waste on resources for more complex tests, which won't run at all if linters fail?

File: .travis.yml [hunk 1]
```diff
@@ -140,6 +140,20 @@ matrix:
...
     after_failure: skip
+  - stage: upload new version of python package to PYPI (only for tagged commits)
+    python: 3.6
+    deploy:
+      provider: pypi
+      server: https://upload.pypi.org/legacy/
+      on:
+        tags: true
+        all_branches: true
+        python: 3.6
```
Comment (jaraco) [inline .travis.yml]: Is it possible to eliminate this selection characteristic now?

Comment (webknjaz) [inline .travis.yml]: Just remove it? Technically, yes. But I would leave it here for safety. We could use YAML anchors to reduce version duplication.

**Pull request merged**